### ISSUE TITLE
Generics through inheritance are *still* unresolved

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/angular_analyzer_plugin#103]
Maybe call this a low-P1. We have a codebase where this is the third most numerous false error, but at this point even the third most common error is only comes up so often.

```
class Generic<T> {
  T something;
}

@Component(...)
class Component extends Generic {
}
```

Using `Component` in a template yields errors

```
<comp [something]="5">...
```

Type int is not assignable to T.

T is an internal type to `Generic`, and has no business being used in the template typechecker. It should be resolved to bounds (in this case `dynamic` since there are no bounds) and inherited constraints before being typechecked.

**Comment by MichaelRFairhurst** [comment]
Merged PR #111
